name.abuchen.portfolio.tests/src/name/abuchen/portfolio/datatransfer/pdf/cetesdirecto/CetesDirectoPDFExtractorTest.java

```diff
@@ -0,0 +1,84 @@
+import name.abuchen.portfolio.model.Client;
+import name.abuchen.portfolio.model.PortfolioTransaction;
+import name.abuchen.portfolio.model.Security;
+import name.abuchen.portfolio.model.Transaction.Unit;
+import name.abuchen.portfolio.money.CurrencyUnit;
+import name.abuchen.portfolio.money.Money;
+import name.abuchen.portfolio.money.Values;
+
+@SuppressWarnings("nls")
+public class CetesDirectoPDFExtractorTest
+{
+
+    @Test
+    public void testEdoCta01()
+    {
+        CetesDirectoPDFExtractor extractor = new CetesDirectoPDFExtractor(new Client());
+
+        List<Exception> errors = new ArrayList<>();
+
+        List<Item> results = extractor.extract(PDFInputFile.loadTestCase(getClass(), "EdoCta01.txt"), errors);
+
+        assertThat(errors, empty());
+        assertThat(countSecurities(results), is(2L));
+        assertThat(countBuySell(results), is(12L));
+        assertThat(countAccountTransactions(results), is(0L));
+        assertThat(results.size(), is(14));
+        new AssertImportActions().check(results, CurrencyUnit.MXN);
+
+        // check security
+        Security security = results.stream().filter(SecurityItem.class::isInstance).findFirst()
+                        .orElseThrow(IllegalArgumentException::new).getSecurity();
+        assertNull(security.getIsin());
+        assertNull(security.getWkn());
+        assertNull(security.getTickerSymbol());
+        assertThat(security.getName(), is("CETES"));
+        assertThat(security.getCurrencyCode(), is(CurrencyUnit.MXN));
+
+        // check buy sell transaction
+        BuySellEntry entry = (BuySellEntry) results.stream().filter(BuySellEntryItem.class::isInstance).findFirst()
+                        .orElseThrow(IllegalArgumentException::new).getSubject();
+
+        assertThat(entry.getPortfolioTransaction().getType(), is(PortfolioTransaction.Type.BUY));
+        assertThat(entry.getAccountTransaction().getType(), is(AccountTransaction.Type.BUY));
+
+        assertThat(entry.getPortfolioTransaction().getDateTime(), is(LocalDateTime.parse("2022-01-06T00:00:00")));
+        assertThat(entry.getPortfolioTransaction().getShares(), is(Values.Share.factorize(6080)));
+        assertThat(entry.getSource(), is("EdoCta01.txt"));
+        assertThat(entry.getNote(), is("ID:SVD147529623 Series:220203 Term:2 Rate:5.51"));
+
+        assertThat(entry.getPortfolioTransaction().getMonetaryAmount(),
+                        is(Money.of(CurrencyUnit.MXN, Values.Amount.factorize(60540.38))));
+        assertThat(entry.getPortfolioTransaction().getGrossValue(),
+                        is(Money.of(CurrencyUnit.MXN, Values.Amount.factorize(60540.38))));
+        assertThat(entry.getPortfolioTransaction().getUnitSum(Unit.Type.TAX),
+                        is(Money.of(CurrencyUnit.MXN, Values.Amount.factorize(0.00))));
+        assertThat(entry.getPortfolioTransaction().getUnitSum(Unit.Type.FEE),
+                        is(Money.of(CurrencyUnit.MXN, Values.Amount.factorize(0.00))));
```

Comment:
We created custom matcher that make these kinds of tests easier to write and easier to read. I have refactored the code accordingly.